staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -141,28 +143,30 @@ func (e *watchEmbeddedEncoder) embeddedIdentifier() runtime.Identifier {
 //
 // NOTE: watchEncoder is NOT thread-safe.
 type watchEncoder struct {
-	ctx             context.Context
-	kind            schema.GroupVersionKind
-	embeddedEncoder runtime.Encoder
-	encoder         runtime.Encoder
-	framer          io.Writer
+	ctx                    context.Context
+	kind                   schema.GroupVersionKind
+	watchObjectsEncoder    runtime.Encoder
+	watchEventsEncoder     runtime.Encoder
+	framer                 io.Writer
+	watchListTransformerFn watchListTransformerFunction
 
 	buffer      runtime.Splice
 	eventBuffer runtime.Splice
 
-	currentEmbeddedIdentifier runtime.Identifier
-	identifiers               map[watch.EventType]runtime.Identifier
+	currentWatchObjectsEncoderIdentifier runtime.Identifier
```

Comment:
Can we avoid renames in this PR?
I wasn't the original proposer of the existing ones, but I actually like them (embeddedEncoder & encoder).

Reply:
without renaming we will end up with embeddedEncoder & encoder & negociatedEncoder

Reply:
I doesn't bother me...